src/duplicacy_chunkmaker.go

```diff
@@ -48,7 +49,7 @@ func CreateChunkMaker(config *Config, hashOnly bool) *ChunkMaker {
 		hashMask:         uint64(config.AverageChunkSize - 1),
 		maximumChunkSize: config.MaximumChunkSize,
 		minimumChunkSize: config.MinimumChunkSize,
-		bufferCapacity:   2 * config.MinimumChunkSize,
+		bufferCapacity:   config.MaximumChunkSize,
```

Comment:
In my case I had to **increase** the circular buffer to `MaximumChunkSize` for the file boundary detection to work properly (it is a while since my implementation so I do not remember the details, but I think it had something to do with not wanting the data to be moved to the chunk too early).

I never thought `MaximumChunkSize` could be smaller than `2 * MinimumChunkSize`, so the correct implementation should take the larger of those values.